vllm/model_executor/layers/rotary_embedding.py

```diff
@@ -453,6 +453,118 @@ def forward(
         return query.flatten(-2), key.flatten(-2)
 
 
+def yarn_get_mscale(scale: float = 1, mscale: float = 1) -> float:
+    if scale <= 1:
+        return 1.0
+    return 0.1 * mscale * math.log(scale) + 1.0
+
+
+class DeepseekScalingRotaryEmbedding(RotaryEmbedding):
```

Comment:
This is extremely similar to `YaRNScalingRotaryEmbedding`, can you extend that one instead to support `mscale`?